t_string! raises an ERROR

Hi,

when testing the t_string! functionality  with an enhanced examples/csr/src/app.rs

```rust
let inc = move |_| set_counter.update(|count| *count += 1);
    
    let count = move || counter.get();
    
    assert_eq!(t_string!(i18n, click_count, count ),"You clicked 0 times");     // New, to test t_string!
    
    view! {
        <p>{t!(i18n, click_count, count)}</p>
        // <p>{t!(i18n, click_count, count = move || counter.get())}</p>
        <button on:click=inc>{t!(i18n, click_to_inc)}</button>

```

 I did experience  the following issue (Note: feature "interpolate_display" has beenenabled in Cargo.toml) :

```shell
error[E0599]: no method named `var_count_string` found for struct `click_count_builder` in the current scope
  --> src/app.rs:34:45
   |
34 |     assert_eq!(t_string!(i18n, click_count, count ),"You clicked 0 times");
   |                                             ^^^^^
   |
  ::: src/main.rs:4:1
   |
4  | leptos_i18n::load_locales!();
   | ---------------------------- method `var_count_string` not found for this struct
   |
help: there is a method `var_count` with a similar name
   |
34 |     assert_eq!(t_string!(i18n, click_count, var_count ),"You clicked 0 times");
   |                                             ~~~~~~~~~

For more information about this error, try `rustc --explain E0599`.
error: could not compile `csr` (bin "csr") due to 1 previous error
2024-05-14T12:09:57.122064Z ERROR ❌ error
```
Any suggestion what I'm doing wrong. Here my toolchain, I'm using most actual i18n version 

```shell
active toolchain
----------------

stable-x86_64-unknown-linux-gnu (default)
rustc 1.78.0 (9b00956e5 2024-04-29)
```

Thank you in advance
BR

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -27,7 +27,7 @@ jobs:
           cargo install cargo-all-features
           cargo test-all-features
 
-  compile_ssr_exemples:
+  compile_ssr_examples:
     name: Compile ${{ matrix.examples }} example
     runs-on: ubuntu-latest
     strategy:
@@ -44,13 +44,13 @@ jobs:
           toolchain: "stable"
           targets: "wasm32-unknown-unknown"
 
-      - name: "Build ${{ matrix.examples }} exemple"
+      - name: "Build ${{ matrix.examples }} example"
         working-directory: examples/${{ matrix.examples }}
         run: |
           cargo install cargo-leptos
           cargo leptos build
 
-  compile_csr_exemples:
+  compile_csr_examples:
     name: Compile ${{ matrix.examples }} example
     runs-on: ubuntu-latest
     strategy:
@@ -76,7 +76,7 @@ jobs:
           toolchain: "stable"
           targets: "wasm32-unknown-unknown"
 
-      - name: "Build ${{ matrix.examples }} exemple"
+      - name: "Build ${{ matrix.examples }} example"
         working-directory: examples/${{ matrix.examples }}
         run: |
           cargo install trunk

--- a/docs/book/src/usage/03_t_macro.md
+++ b/docs/book/src/usage/03_t_macro.md
@@ -201,3 +201,7 @@ To avoid confusion with subkeys you can use `::` to separate the namespace name 
 ```rust
 t!(i18n, my_namespace::hello_world)
 ```
+
+## `tu!`
+
+the `tu!` macro is the same as `t!` but untracked.

--- a/leptos_i18n/src/lib.rs
+++ b/leptos_i18n/src/lib.rs
@@ -140,10 +140,10 @@ pub use locale_traits::*;
 
 pub use context::{provide_i18n_context, use_i18n_context, I18nContext};
 
-pub use leptos_i18n_macro::{load_locales, t, td};
+pub use leptos_i18n_macro::{load_locales, t, td, tu};
 
 #[cfg(feature = "interpolate_display")]
-pub use leptos_i18n_macro::{t_display, t_string, td_display, td_string};
+pub use leptos_i18n_macro::{t_display, t_string, td_display, td_string, tu_display, tu_string};
 
 #[doc(hidden)]
 pub mod __private {

--- a/leptos_i18n_macro/src/lib.rs
+++ b/leptos_i18n_macro/src/lib.rs
@@ -64,6 +64,12 @@ pub fn t(tokens: proc_macro::TokenStream) -> proc_macro::TokenStream {
     t_macro::t_macro(tokens, InputType::Context, OutputType::View)
 }
 
+/// Same as the `t!` macro but untracked.
+#[proc_macro]
+pub fn tu(tokens: proc_macro::TokenStream) -> proc_macro::TokenStream {
+    t_macro::t_macro(tokens, InputType::Untracked, OutputType::View)
+}
+
 /// Just like the `t!` macro but instead of taking `I18nContext` as the first argument it takes the desired locale.
 ///
 /// Usage:
@@ -104,10 +110,15 @@ pub fn td(tokens: proc_macro::TokenStream) -> proc_macro::TokenStream {
 ///     "You clicked a lot of times"
 /// )
 /// ```
-#[cfg(feature = "interpolate_display")]
 #[proc_macro]
 pub fn t_string(tokens: proc_macro::TokenStream) -> proc_macro::TokenStream {
     t_macro::t_macro(tokens, InputType::Context, OutputType::String)
+}
+
+/// Same as the `t_string!` macro but untracked.
+#[proc_macro]
+pub fn tu_string(tokens: proc_macro::TokenStream) -> proc_macro::TokenStream {
+    t_macro::t_macro(tokens, InputType::Untracked, OutputType::String)
 }
 
 /// Just like the `t_string!` macro but return either a struct implementing `Display` or a `&'static str` instead of a `Cow<'static, str>`.
@@ -130,10 +141,15 @@ pub fn t_string(tokens: proc_macro::TokenStream) -> proc_macro::TokenStream {
 ///
 /// assert_eq!(t_str, "You clicked 10 times");
 /// ```
-#[cfg(feature = "interpolate_display")]
 #[proc_macro]
 pub fn t_display(tokens: proc_macro::TokenStream) -> proc_macro::TokenStream {
     t_macro::t_macro(tokens, InputType::Context, OutputType::Display)
+}
+
+/// Same as the `t_display!` macro but untracked.
+#[proc_macro]
+pub fn tu_display(tokens: proc_macro::TokenStream) -> proc_macro::TokenStream {
+    t_macro::t_macro(tokens, InputType::Untracked, OutputType::Display)
 }
 
 /// Just like the `t_string!` macro but takes the `Locale` as an argument instead of the context.
@@ -154,7 +170,6 @@ pub fn t_display(tokens: proc_macro::TokenStream) -> proc_macro::TokenStream {
 ///     "You clicked a lot of times"
 /// )
 /// ```
-#[cfg(feature = "interpolate_display")]
 #[proc_macro]
 pub fn td_string(tokens: proc_macro::TokenStream) -> proc_macro::TokenStream {
     t_macro::t_macro(tokens, InputType::Locale, OutputType::String)
@@ -178,7 +193,6 @@ pub fn td_string(tokens: proc_macro::TokenStream) -> proc_macro::TokenStream {
 ///
 /// assert_eq!(t_str, "You clicked 10 times");
 /// ```
-#[cfg(feature = "interpolate_display")]
 #[proc_macro]
 pub fn td_display(tokens: proc_macro::TokenStream) -> proc_macro::TokenStream {
     t_macro::t_macro(tokens, InputType::Locale, OutputType::Display)

--- a/leptos_i18n_macro/src/load_locales/interpolate.rs
+++ b/leptos_i18n_macro/src/load_locales/interpolate.rs
@@ -1,10 +1,8 @@
 use std::collections::HashSet;
 
 use proc_macro2::{Span, TokenStream};
-use quote::quote;
-
-#[cfg(feature = "interpolate_display")]
 use quote::format_ident;
+use quote::quote;
 
 use super::{
     key::Key,
@@ -12,7 +10,6 @@ use super::{
     parsed_value::{InterpolateKey, ParsedValue},
 };
 
-#[cfg(feature = "debug_interpolations")]
 const MAX_KEY_GENERATE_BUILD_DEBUG: usize = 4;
 
 pub struct Interpolation {
@@ -25,7 +22,6 @@ struct Field<'a> {
     generic: syn::Ident,
     name: String,
     kind: &'a InterpolateKey,
-    #[cfg(feature = "debug_interpolations")]
     real_name: &'a str,
 }
 
@@ -45,7 +41,6 @@ impl Interpolation {
         let fields = keys_set
             .iter()
             .map(|kind| {
-                #[cfg(feature = "debug_interpolations")]
                 let real_name = kind.get_real_name();
                 let key = kind
                     .as_key()
@@ -57,7 +52,6 @@ impl Interpolation {
                     generic,
                     name,
                     kind,
-                    #[cfg(feature = "debug_interpolations")]
                     real_name,
                 }
             })
@@ -69,11 +63,11 @@ impl Interpolation {
             Self::into_view_impl(key, &ident, &locale_field, &fields, locales, default_match);
         let debug_impl = Self::debug_impl(&builder_name, &ident, &fields);
 
-        #[cfg(feature = "interpolate_display")]
-        let display_impl =
-            Self::display_impl(key, &ident, &locale_field, &fields, locales, default_match);
-        #[cfg(not(feature = "interpolate_display"))]
-        let display_impl = quote!();
+        let display_impl = if cfg!(feature = "interpolate_display") {
+            Self::display_impl(key, &ident, &locale_field, &fields, locales, default_match)
+        } else {
+            quote!()
+        };
 
         let new_impl = Self::new_impl(&ident, &locale_field, &fields);
         let default_generics = fields
@@ -136,7 +130,6 @@ impl Interpolation {
             .collect()
     }
 
-    #[cfg(feature = "debug_interpolations")]
     fn generate_build_fns(ident: &syn::Ident, fields: &[Field], locale_field: &Key) -> TokenStream {
         if fields.len() > MAX_KEY_GENERATE_BUILD_DEBUG {
             return Self::generate_success_build_fn(ident, fields);
@@ -151,7 +144,6 @@ impl Interpolation {
         }
     }
 
-    #[cfg(feature = "interpolate_display")]
     fn generate_string_build(ident: &syn::Ident, fields: &[Field]) -> TokenStream {
         let left_generics_string = fields.iter().filter_map(|field| {
             let ident = &field.generic;
@@ -194,11 +186,11 @@ impl Interpolation {
             quote!(#ident)
         });
 
-        #[cfg(feature = "interpolate_display")]
-        let string_build = Self::generate_string_build(ident, fields);
-
-        #[cfg(not(feature = "interpolate_display"))]
-        let string_build = quote!();
+        let string_build = if cfg!(feature = "interpolate_display") {
+            Self::generate_string_build(ident, fields)
+        } else {
+            quote!()
+        };
 
         quote! {
 
@@ -218,10 +210,11 @@ impl Interpolation {
     fn builder_impl(ident: &syn::Ident, locale_field: &Key, fields: &[Field]) -> TokenStream {
         let set_fns = Self::genenerate_set_fns(ident, locale_field, fields);
 
-        #[cfg(not(feature = "debug_interpolations"))]
-        let build_fns = Self::generate_success_build_fn(ident, fields);
-        #[cfg(feature = "debug_interpolations")]
-        let build_fns = Self::generate_build_fns(ident, fields, locale_field);
+        let build_fns = if cfg!(not(feature = "debug_interpolations")) {
+            Self::generate_success_build_fn(ident, fields)
+        } else {
+            Self::generate_build_fns(ident, fields, locale_field)
+        };
 
         quote! {
             #set_fns
@@ -263,7 +256,6 @@ impl Interpolation {
             .chain(right_fields.iter().map(other_field_map_fn))
     }
 
-    #[cfg(feature = "debug_interpolations")]
     fn generate_default_constructed(
         ident: &syn::Ident,
         fields: &[Field],
@@ -287,7 +279,6 @@ impl Interpolation {
         }
     }
 
-    #[cfg(feature = "debug_interpolations")]
     fn generate_all_failing_build_fn<'a>(
         ident: &'a syn::Ident,
         fields: &'a [Field],
@@ -309,7 +300,6 @@ impl Interpolation {
         })
     }
 
-    #[cfg(feature = "debug_interpolations")]
     fn generate_failing_build_fn<'a, I>(
         self_ident: &syn::Ident,
         default_constructed: &TokenStream,
@@ -375,18 +365,27 @@ impl Interpolation {
             quoted_gen,
         );
 
-        #[cfg(feature = "interpolate_display")]
-        let output_field_generic_string = match field.kind.get_string_generic() {
-            Ok(bounds) => quote!(impl #bounds),
-            Err(t) => quote!(#t),
-        };
-        #[cfg(feature = "interpolate_display")]
-        let output_generics_string = Self::generate_generics(
-            left_fields,
-            Some(output_field_generic_string.clone()),
-            right_fields,
-            quoted_gen,
-        );
+        let (output_field_generic_string, output_generics_string, bounds) =
+            if cfg!(feature = "interpolate_display") {
+                let (output_field_generic_string, bounds) = match field.kind.get_string_generic() {
+                    Ok(bounds) => (quote!(impl #bounds), bounds),
+                    Err(t) => (quote!(#t), quote!()),
+                };
+                let output_generics_string = Self::generate_generics(
+                    left_fields,
+                    Some(output_field_generic_string.clone()),
+                    right_fields,
+                    quoted_gen,
+                );
+                (
+                    output_field_generic_string,
+                    Some(output_generics_string),
+                    bounds,
+                )
+            } else {
+                (quote!(), None, quote!())
+            };
+
         let other_fields = Self::generate_generics(left_fields, None, right_fields, |field| {
             if let Some(key) = field.kind.as_key() {
                 quote!(#key)
@@ -404,7 +403,7 @@ impl Interpolation {
         };
         let restructure = quote!(#ident { #(#other_fields,)* #kind });
 
-        let fns = match kind {
+        let (set_function, set_str_function) = match kind {
             InterpolateKey::Variable(key) => (
                 quote! {
                     #[inline]
@@ -415,8 +414,7 @@ impl Interpolation {
                         #restructure
                     }
                 },
-                #[cfg(feature = "interpolate_display")]
-                {
+                if let Some(output_generics_string) = output_generics_string {
                     let string_key = format_ident!("{}_string", key.ident);
                     quote! {
                         #[inline]
@@ -426,6 +424,8 @@ impl Interpolation {
                             #restructure
                         }
                     }
+                } else {
+                    quote!()
                 },
             ),
             InterpolateKey::Component(key) => (
@@ -441,8 +441,7 @@ impl Interpolation {
                         #restructure
                     }
                 },
-                #[cfg(feature = "interpolate_display")]
-                {
+                if let Some(output_generics_string) = output_generics_string {
                     let string_key = format_ident!("{}_string", key.ident);
                     quote! {
                         #[inline]
@@ -452,6 +451,8 @@ impl Interpolation {
                             #restructure
                         }
                     }
+                } else {
+                    quote!()
                 },
             ),
             InterpolateKey::Count(plural_type) => (
@@ -464,22 +465,20 @@ impl Interpolation {
                         #restructure
                     }
                 },
-                #[cfg(feature = "interpolate_display")]
-                quote! {
-                    #[inline]
+                if let Some(output_generics_string) = output_generics_string {
+                    quote! {
+                        #[inline]
                     pub fn var_count_string(self, var_count: #plural_type) -> #ident<#(#output_generics_string,)*>
                     {
                         #destructure
                         #restructure
                     }
+                    }
+                } else {
+                    quote!()
                 },
             ),
         };
-
-        #[cfg(feature = "interpolate_display")]
-        let (set_function, set_str_function) = fns;
-        #[cfg(not(feature = "interpolate_display"))]
-        let (set_function, set_str_function) = (fns.0, quote!());
 
         if cfg!(feature = "debug_interpolations") {
             let left_generics_empty =
@@ -512,6 +511,35 @@ impl Interpolation {
                 }
             };
 
+            let string_impl = if cfg!(feature = "interpolate_display") {
+                let left_generics_empty_string = left_generics_empty.clone();
+                let right_generics_empty_string = right_generics_empty.clone();
+                let left_generics_already_set_string = Self::generate_generics(
+                    left_fields,
+                    Some({
+                        let field_gen = &field.generic;
+                        quote!(#field_gen: #bounds)
+                    }),
+                    right_fields,
+                    quoted_gen,
+                );
+                let right_generics_already_set_string = right_generics_already_set.clone();
+                quote! {
+                    #[allow(non_camel_case_types)]
+                    impl<#(#left_generics_empty_string,)*> #ident<#(#right_generics_empty_string,)*> {
+                        #set_str_function
+                    }
+
+                    #[allow(non_camel_case_types)]
+                    impl<#(#left_generics_already_set_string,)*> #ident<#(#right_generics_already_set_string,)*> {
+                        #[deprecated(note = #compile_warning)]
+                        #set_str_function
+                    }
+                }
+            } else {
+                quote!()
+            };
+
             quote! {
                 #[allow(non_camel_case_types)]
                 impl<#(#left_generics_empty,)*> #ident<#(#right_generics_empty,)*> {
@@ -522,6 +550,8 @@ impl Interpolation {
                     #[deprecated(note = #compile_warning)]
                     #set_function
                 }
+
+                #string_impl
             }
         } else {
             let left_generics =
@@ -556,7 +586,6 @@ impl Interpolation {
         }
     }
 
-    #[cfg(feature = "interpolate_display")]
     fn display_impl(
         key: &Key,
         ident: &syn::Ident,
@@ -669,7 +698,6 @@ impl Interpolation {
             })
     }
 
-    #[cfg(feature = "interpolate_display")]
     fn create_locale_string_impl<'a>(
         key: &'a Key,
         locales: &'a [Locale],

--- a/leptos_i18n_macro/src/load_locales/mod.rs
+++ b/leptos_i18n_macro/src/load_locales/mod.rs
@@ -60,11 +60,13 @@ pub fn load_locales() -> Result<TokenStream> {
 
     let file_tracking = tracking::generate_file_tracking();
 
-    let mut macros_reexport = vec![quote!(t), quote!(td)];
+    let mut macros_reexport = vec![quote!(t), quote!(td), quote!(tu)];
     if cfg!(feature = "interpolate_display") {
         macros_reexport.extend([
             quote!(t_string),
+            quote!(tu_string),
             quote!(t_display),
+            quote!(tu_display),
             quote!(td_string),
             quote!(td_display),
         ]);

--- a/leptos_i18n_macro/src/load_locales/parsed_value.rs
+++ b/leptos_i18n_macro/src/load_locales/parsed_value.rs
@@ -558,7 +558,6 @@ impl ParsedValue {
         }
     }
 
-    #[cfg(feature = "interpolate_display")]
     fn flatten_string(&self, tokens: &mut Vec<TokenStream>) {
         match self {
             ParsedValue::Subkeys(_) | ParsedValue::Default => {}
@@ -584,7 +583,6 @@ impl ParsedValue {
         }
     }
 
-    #[cfg(feature = "interpolate_display")]
     pub fn as_string_impl(&self) -> TokenStream {
         let mut tokens = Vec::new();
         self.flatten_string(&mut tokens);
@@ -635,7 +633,6 @@ impl InterpolateKey {
         }
     }
 
-    #[cfg(feature = "debug_interpolations")]
     pub fn get_real_name(&self) -> &str {
         match self {
             InterpolateKey::Count(_) => "count",
@@ -660,7 +657,6 @@ impl InterpolateKey {
         }
     }
 
-    #[cfg(feature = "interpolate_display")]
     pub fn get_string_generic(&self) -> Result<TokenStream, PluralType> {
         match self {
             InterpolateKey::Count(t) => Err(*t),
@@ -669,7 +665,6 @@ impl InterpolateKey {
         }
     }
 
-    #[cfg(feature = "debug_interpolations")]
     pub fn get_default(&self) -> TokenStream {
         match self {
             InterpolateKey::Variable(_) => {

--- a/leptos_i18n_macro/src/load_locales/plural.rs
+++ b/leptos_i18n_macro/src/load_locales/plural.rs
@@ -100,7 +100,6 @@ pub enum Plurals {
 }
 
 impl Plurals {
-    #[cfg(feature = "interpolate_display")]
     pub fn as_string_impl(&self) -> TokenStream {
         match self {
             Plurals::I8(plurals) => Self::to_tokens_integers_string(plurals),
@@ -253,7 +252,6 @@ impl Plurals {
         }
     }
 
-    #[cfg(feature = "interpolate_display")]
     fn to_tokens_integers_string<T: PluralInteger>(
         plurals: &[(Plural<T>, ParsedValue)],
     ) -> TokenStream {
@@ -326,7 +324,6 @@ impl Plurals {
         }
     }
 
-    #[cfg(feature = "interpolate_display")]
     fn to_tokens_floats_string<T: PluralFloats>(
         plurals: &[(Plural<T>, ParsedValue)],
     ) -> TokenStream {

--- a/leptos_i18n_macro/src/t_macro/mod.rs
+++ b/leptos_i18n_macro/src/t_macro/mod.rs
@@ -12,9 +12,7 @@ pub mod parsed_input;
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
 pub enum OutputType {
     View,
-    #[cfg(feature = "interpolate_display")]
     String,
-    #[cfg(feature = "interpolate_display")]
     Display,
 }
 
@@ -22,6 +20,7 @@ pub enum OutputType {
 pub enum InputType {
     Locale,
     Context,
+    Untracked,
 }
 
 pub fn t_macro(
@@ -91,9 +90,7 @@ impl OutputType {
     pub fn build_fn(self) -> TokenStream {
         match self {
             OutputType::View => quote!(build),
-            #[cfg(feature = "interpolate_display")]
             OutputType::String => quote!(build_string),
-            #[cfg(feature = "interpolate_display")]
             OutputType::Display => quote!(build_display),
         }
     }
@@ -101,7 +98,6 @@ impl OutputType {
     pub fn is_string(self) -> bool {
         match self {
             OutputType::View => false,
-            #[cfg(feature = "interpolate_display")]
             OutputType::String | OutputType::Display => true,
         }
     }
@@ -114,7 +110,6 @@ impl OutputType {
                     move || #ts
                 }
             },
-            #[cfg(feature = "interpolate_display")]
             OutputType::String | OutputType::Display => quote! {
                 {
                     #params
@@ -129,6 +124,9 @@ impl InputType {
     pub fn get_key<T: ToTokens>(self, input: T, keys: Keys) -> TokenStream {
         match self {
             InputType::Context => quote!(leptos_i18n::I18nContext::get_keys(#input).#keys),
+            InputType::Untracked => {
+                quote!(leptos_i18n::I18nContext::get_keys_untracked(#input).#keys)
+            }
             InputType::Locale => quote!(leptos_i18n::Locale::get_keys(#input).#keys),
         }
     }

